frontend/src/components/forms/CreateResident.tsx

```diff
@@ -61,17 +63,30 @@ const CreateResident = ({
   const [isOpen, setIsOpen] = useState(false);
   const [showAlert, setShowAlert] = useState(false);
 
+  const newToast = CreateToast();
+
   const ROOM_ERROR_TEXT = `Room Number is required and must only contain numbers.`;
   const addResident = async () => {
-    await ResidentAPIClient.createResident({
+    const res = await ResidentAPIClient.createResident({
       initial: initials.toUpperCase(),
       roomNum: parseInt(roomNumber, 10),
       dateJoined: convertToString(moveInDate),
       buildingId,
     });
-    getRecords(1);
-    countResidents();
-    setUserPageNum(1);
+
+    if (isResidentErrorResponse(res)) {
+      newToast(
+        "Error creating resident",
+        res.errMessage,
+        "error"
+      )
+    }
+    else if (res) {
+      getRecords(1);
+      countResidents();
+      setUserPageNum(1);
+      setShowAlert(true)
```

Comment:
We move the call to here since the showAlert state controls the visibility of a `success` alert ; we only want the success state to be displayed if we get back a boolean result with value `true`